overlord/servicestate/quota_control.go

```diff
@@ -266,6 +292,38 @@ func UpdateQuota(st *state.State, name string, updateOpts QuotaGroupUpdate) (*st
 	return ts, nil
 }
 
+// remove a string item at index i from the string slice,
+// it maintains the ordering of the original slice.
+func remove(slice []string, i int) []string {
+	return append(slice[:i], slice[i+1:]...)
+}
+
+// ensureSnapServicesAbsentFromSubGroups removes all service references of a snap in
+// sub-groups related to the group of the snap, and returns the groups that were modified.
+func ensureSnapServicesAbsentFromSubGroups(grp *quota.Group, snap string, allGrps map[string]*quota.Group) ([]*quota.Group, error) {
```

Comment:
nitpick: `removeServicesFromSubGroups`

Reply:
Like this suggestion much better, thanks!